### PR TITLE
Create list-container component as a replacement for current CRUD helpers

### PR DESCRIPTION
**Pull request by xantorres**

### CODE REVIEW

File: src/app/plugins/kubernetes/components/storage/storage-classes-picklist.tsx [hunk 1]
```diff
@@ -32,10 +35,16 @@ const StorageClassesPicklist: React.ComponentType<Props> = forwardRef<HTMLElemen
       showNone = true,
       ...rest
     } = props
-
-    const [storageClasses, storageClassesLoading] = useDataLoader(storageClassActions.list, {
-      clusterId,
-    }) as any
+    const params = useMemo(
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/storage-classes-picklist.tsx:38]: do we still need to `useMemo` on params? I thought we addressed this internally with memoization

File: src/app/core/containers/list-container.tsx [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+      {!DeleteDialog && handleRemove && showingConfirmDialog && (
+        <ConfirmationDialog
+          open={showingConfirmDialog}
+          text={deleteConfirmText}
+          onCancel={toggleConfirmDialog}
+          onConfirm={handleDeleteConfirm}
+        />
+      )}
+      {addEnabled && (
+        <PageContainerHeader>
```
Comment (jruddell) [inline src/app/core/containers/list-container.tsx:209]: We should either update the PageContainer to integrate with the portals or just remove PageContainer completely. If we integrate then we should be really consistent with using this component to update header elements in the page

File: src/app/plugins/kubernetes/components/storage/storage-classes/list-page.tsx [hunk 3]
```diff
@@ -77,12 +42,35 @@ export const options = {
...
-  cacheKey: ActionDataKeys.StorageClasses,
-  actions: storageClassActions,
   name: 'StorageClasses',
   title: 'Storage Classes',
-  ListPage,
 }
 
-const components = createCRUDComponents(options)
-export default components.ListPage
+const ListPage = () => {
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/storage-classes/list-page.tsx:49]: 😍

**Review by jruddell — APPROVED** [review]

File: src/app/plugins/kubernetes/components/storage/storage-classes-picklist.tsx (hunk 1 above)
Comment (xantorres) [reply src/app/plugins/kubernetes/components/storage/storage-classes-picklist.tsx:38]: no, we don't need that anymore

**Pull request closed without merging**